plugins/modules/openssl_csr_pipe.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2017, Yanis Guenane <yanis+ansible@guenane.org>
+# Copyright: (c) 2020, Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: openssl_csr_pipe
+short_description: Generate OpenSSL Certificate Signing Request (CSR)
+version_added: 1.3.0
+description:
+    - "Please note that the module regenerates existing CSR if it doesn't match the module's
+      options, or if it seems to be corrupt."
+author:
+- Yanis Guenane (@Spredzy)
+- Felix Fontein (@felixfontein)
+options:
+    content:
+        description:
+            - The existing CSR.
+        type: str
+extends_documentation_fragment:
+- community.crypto.module_csr
+seealso:
+- module: community.crypto.openssl_csr
+'''
+
+EXAMPLES = r'''
+- name: Generate an OpenSSL Certificate Signing Request
+  community.crypto.openssl_csr_pipe:
+    privatekey_path: /etc/ssl/private/ansible.com.pem
+    common_name: www.ansible.com
+  register: result
+- debug:
+    var: result.csr
+
+- name: Generate an OpenSSL Certificate Signing Request with an inline key
+  community.crypto.openssl_csr:
+    content: "{{ lookup('file', '/etc/ssl/csr/www.ansible.com.csr') }}"
+    privatekey_content: "{{ private_key_content }}"
+    common_name: www.ansible.com
+  register: result
+- name: Store CSR
+  ansible.builtin.copy:
+    path: /etc/ssl/csr/www.ansible.com.csr
+    content: "{{ result.csr }}"
+  when: result is changed
+'''
+
+RETURN = r'''
+privatekey:
```

Comment:
```suggestion
privatekey_path:
```

Not sure if this is more consistent, but the suffix `_path` and `_content` might make it clearer in many cases what's what.

Reply:
That would be a breaking change, since the return value has always been `privatekey` until now.

Reply:
Ah, sorry, this is new, so it's not really a breaking change. It is something returned by the common code of openssl_privatekey and openssl_privatekey_pipe though, so if we name the return value differently here, we'd have to introduce some special casing. Not sure whether that's worth the trouble.

Reply:
Probably better left for some eventual refactoring then.